### ISSUE TITLE
Blank preview on M1

### ISSUE THREAD

**Issue by manuelgrabowski** [whomwah/qlstephen#133]
I'm afraid the plugin doesn't work on my M1, while the same file/version works fine on my Intel Mac. Judging by the comments in https://github.com/sindresorhus/quick-look-plugins/issues/126 I'm guessing this is an M1 issue, but I'm happy to provide any additional debug information to narrow it down further.

![image](https://user-images.githubusercontent.com/13483728/148806022-fcd96c9f-0239-46b0-8d3c-228e2d652d22.png)

**Comment by philfreo** [comment]
I have a new M1 Mac I'm trying to setup qlstephen on. I'm unable to get it working at all, even after following all the same steps as in https://github.com/whomwah/qlstephen/issues/87#issuecomment-590572347. I don't have the blank screen issue, I have the ql plugin taking no effect whatsoever.
Is Rosetta required? (I don't have that installed currently)

**Comment by SConaway** [comment]
yes, as of now, it's compiled only for x86_64, so Rosetta is needed.

**Comment by SConaway** [comment]
and it seems like running `make && make install` in the repo (after cloning) compiles it for arm64 as well

**Comment by grischard** [comment]
For those too lazy to recompile, here's a fresh build: [QLStephen.qlgenerator.zip](https://github.com/whomwah/qlstephen/files/8051885/QLStephen.qlgenerator.zip)

**Comment by philfreo** [comment]
Is it possible to get the homebrew version updated for M1 as well?

**Comment by SConaway** [comment]
As far as I know, brew doesn't accept forks officially. I'm not sure what's happening with this repo...

**Comment by bengry** [comment]
> For those too lazy to recompile, here's a fresh build: [QLStephen.qlgenerator.zip](https://github.com/whomwah/qlstephen/files/8051885/QLStephen.qlgenerator.zip)

Does this fix the empty quicklook previews for you? I tried replacing the file I had in `~/Library/QuickLook` and they still don't work, even after a restart. I should note I'm getting the same empty preview even using my own compiled version (compiled on an M1 using the instructions in the README).

**Comment by davidjdixon** [comment]
> > For those too lazy to recompile, here's a fresh build: [QLStephen.qlgenerator.zip](https://github.com/whomwah/qlstephen/files/8051885/QLStephen.qlgenerator.zip)
> 
> Does this fix the empty quicklook previews for you? I tried replacing the file I had in `~/Library/QuickLook` and they still don't work, even after a restart. I should note I'm getting the same empty preview even using my own compiled version (compiled on an M1 using the instructions in the README).

I can confirm that this update does fix the issue I was seeing. But to clarify, my issue wasn't an empty preview, just no preview at all, so I would initially check if the file has the quarantine attributes.

Unfortunately, it seems whatever version of the binary that homebrew is using is still not arm compatible (as the original version was only installed yesterday during a system rebuild).

**Comment by brettrowberry** [comment]
Looking forward to this on M1 Homebrew as well!

**Comment by justinormont** [comment]
> For those too lazy to recompile, here's a fresh build: [QLStephen.qlgenerator.zip](https://github.com/whomwah/qlstephen/files/8051885/QLStephen.qlgenerator.zip)

Neither the arm build by [bengry](https://github.com/bengry), nor removing the quarantine flag (`xattr -r -d com.apple.quarantine ~/Library/QuickLook`) helped. Restarting quicklook (`qlmanage -r`) and Finder (`killall Finder`) also did not help.

Any other ideas to fix the blank preview?

**Comment by justinormont** [comment]
This Quick Look plugin works well, and is kept up to date -- https://github.com/sbarex/SourceCodeSyntaxHighlight

Can either download from the releases, or use Homebrew `brew install --cask --no-quarantine syntax-highlight`.

**Comment by MarkBTomlinson** [comment]
> This Quick Look plugin works well, and is kept up to date -- https://github.com/sbarex/SourceCodeSyntaxHighlight
> 
> Can either download from the releases, or use Homebrew `brew install --cask --no-quarantine syntax-highlight`.

Thanks for this the other solutions got me nowhere.
